packages/dma-contracts/contracts/actions/common/PullTokenMaxAmount.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.15;
+
+import { Executable } from "../common/Executable.sol";
+import { SafeERC20, IERC20 } from "../../libs/SafeERC20.sol";
+import { PullTokenData } from "../../core/types/Common.sol";
+import "../../core/types/Common.sol";
+
+/**
+ * @title PullToken Action contract
+ * @notice Pulls token from a target address to the current calling context
+ */
+contract PullTokenMaxAmount is Executable {
+  using SafeERC20 for IERC20;
+
+  /**
+   * @dev Is intended to pull tokens in to a user's proxy (the calling context)
+   * @param data Encoded calldata that conforms to the PullTokenData struct
+   */
+  function execute(bytes calldata data, uint8[] memory) external payable override {
+    PullTokenData memory pull = parseInputs(data);
+
+    IERC20 token = IERC20(pull.asset);
+
+    if (pull.amount == type(uint256).max) {
```

Comment:
we should never use that action in automation, I can see possible ways of malicious usage.